node/v2/header.js

```diff
@@ -78,6 +78,13 @@ HeaderRW.prototype.writeInto = function writeInto(headers, buffer, offset) {
         if (res.err) return res;
         offset = res.offset;
 
+        if (typeof headers[key] !== 'string') {
+            return errors.InvalidHeaderTypeError({
```

Comment:
Also, github won't let me comment on it, but you want to do the same check in `HeaderRW#byteLength` above circa L57; there it'll be `LengthResult.error(....)`.

Finally, the fact that any of this didn't fail a test is a bug, please cover these paths and prove they work.